src/libsystemd/sd-journal/journal-file.c

```diff
@@ -2879,6 +2879,7 @@ enum {
                              * should exist in an earlier position. */
         TEST_GOTO_NEXT,     /* No matching object exists in this array and earlier arrays, go to the next array. */
         TEST_GOTO_PREVIOUS, /* No matching object exists in this array and later arrays, go to the previous array. */
+        TEST_CORRUPTED,     /* Found a corrupted entry. */
```

Comment:
There's a long comment a the beginning of `generic_array_bisect_step` that should be expanded with the documentation of this new flag.